app/main.py

```diff
@@ -96,3 +114,47 @@ async def get_subtype(
                 raise HTTPException(status_code=404, detail="Subtype not found")
 
     return {"message": result}
+
+@app.post("/signup", tags=["User"], status_code=201)
+async def signup(
+    cm: Annotated[ConnectionManager, Depends(get_connection_manager)],
+    credentials: HTTPBasicCredentials = Depends(auth)
+):
+    username = credentials.username
+
+    if not username:
+        raise HTTPException(status_code=HTTPStatus.BAD_REQUEST, detail="Missing email address!")
+
+    try:
+        with cm as connection_manager:
+            with connection_manager.get_cursor() as cursor:
+                logger.info(f"Creating user: {username}")
+                user = await new_user(cursor, username, FERTISCAN_STORAGE_URL)
+        return {"user_id": user.get_id()}
+
+    except Exception as e:
+        logger.error(f"Error occurred: {e}")
+        logger.error("Traceback: " + traceback.format_exc())
+        raise HTTPException(status_code=HTTPStatus.INTERNAL_SERVER_ERROR, detail="Failed to create user!")
+
+@app.post("/login", tags=["User"], status_code=200)
+async def login(
+    cm: Annotated[ConnectionManager, Depends(get_connection_manager)],
+    credentials: HTTPBasicCredentials = Depends(auth)
+):
+    username = credentials.username
+    # password = credentials.password()
+
+    if not username:
+        raise HTTPException(status_code=HTTPStatus.BAD_REQUEST, detail="Missing email address!")
+    
+    try:
+        with cm as connection_manager:
+            with connection_manager.get_cursor() as cursor:
+                logger.info(f"Fetching user ID for username: {username}")
+                user = await get_user(cursor, username)
+                return {"user_id": user.get_id()}
+    except Exception as e:
+        logger.error(f"Error occurred: {e}")
+        logger.error("Traceback: " + traceback.format_exc())
+        return HTTPException(status_code=HTTPStatus.INTERNAL_SERVER_ERROR, detail=str(e))
```

Comment:
just use cm directly instead of creating a connection_manager alias, or set the variable name in the function signature to connection_manager directly

Reply:
Same here.